### PR TITLE
Compatiblity with Rails 5.0

### PR DESCRIPTION
**Pull request by pschambacher**
:octopus::computer:

/cc @grosser 

### Steps to merge
* [ ] :+1: from the team

### Risks
* Low: almost every change is done behind an ActiveRecord version check

### CODE REVIEW

File: lib/kasket/relation_mixin.rb [hunk 1]
```diff
@@ -2,9 +2,13 @@ module Kasket
   module RelationMixin
     def to_kasket_query(binds = nil)
       if arel.is_a?(Arel::SelectManager)
-        arel.to_kasket_query(klass, (binds || bind_values))
+        if ActiveRecord::VERSION::MAJOR < 5
+          arel.to_kasket_query(klass, (binds || bind_values))
+        else
+          arel.to_kasket_query(klass, (@values[:where].binds.map(&:value_for_database) + Array(@values[:limit])))
```
Comment (pschambacher) [inline lib/kasket/relation_mixin.rb]: This looks extremely dodgy... really not sure...

Comment (grosser) [inline lib/kasket/relation_mixin.rb]: maybe blame bind_values inide of the old version of arel and see what they did when they removed the method

Comment (grosser) [inline lib/kasket/relation_mixin.rb]: 'd guess it's missing a few things ... like order/group statements etc

File: test/database.yml [hunk 2]
```diff
@@ -2,6 +2,6 @@ test:
   adapter: mysql2
   encoding: utf8
   database: kasket_test
-  username: root
-  password:
-  host: 127.0.0.1
+  username: admin
+  password: 123456
+  host: 192.168.42.45
```
Comment (grosser) [inline test/database.yml]: use ENV vars ?

Comment (pschambacher) [inline test/database.yml]: Ah I should have cleaned after myself. Will add Phenix later.

**Comment by pschambacher** [comment]
Splited a bit this huge PR into smaller bits and pieces.

File: lib/kasket/relation_mixin.rb [hunk 3]
```diff
@@ -2,7 +2,11 @@ module Kasket
   module RelationMixin
     def to_kasket_query(binds = nil)
       if arel.is_a?(Arel::SelectManager)
-        arel.to_kasket_query(klass, (binds || bind_values))
+       if ActiveRecord::VERSION::MAJOR < 5
+         arel.to_kasket_query(klass, (binds || bind_values))
+       else
+         arel.to_kasket_query(klass, (@values[:where].binds.map(&:value_for_database) + Array(@values[:limit])))
```
Comment (pschambacher) [inline lib/kasket/relation_mixin.rb]: OK so regarding this line. Kasket actually doesn't support [with](https://github.com/zendesk/kasket/blob/master/lib/kasket/visitor.rb#L29), [offset](https://github.com/zendesk/kasket/blob/master/lib/kasket/visitor.rb#L30), [order](https://github.com/zendesk/kasket/blob/master/lib/kasket/visitor.rb#L32), [group by](https://github.com/zendesk/kasket/blob/master/lib/kasket/visitor.rb#L47) or [having](https://github.com/zendesk/kasket/blob/master/lib/kasket/visitor.rb#L48).

So it's not worth looking for anything else than some binding in `where` or the `limit`. So despite being awful, this line looks actually to be right.

File: .travis.yml [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
-      gemfile: gemfiles/rails3.2.gemfile
-  allow_failures:
-    - rvm: 2.0.0
-      gemfile: gemfiles/rails4.1.gemfile
-    - rvm: 2.0.0
-      gemfile: gemfiles/rails4.2.gemfile
+    - rvm: 2.1.8
+      gemfile: gemfiles/rails5.0.gemfile
+    - rvm: jruby
+      gemfile: gemfiles/rails5.0.gemfile
```
Comment (grosser) [inline .travis.yml]: also fun: `fast_finish: true`

**Comment by grosser** [comment]
:+1:

**Pull request merged**